File: ios/FamETC/Features/Trips/TripsScreen.swift
import SwiftUI

/// The Trips webview surface (docs/TRIPS-PLAN.md "iOS" §3): itinerary,
/// flights, lodging, invites, and the trips list live on the web at `/trips`
/// (see the plan's "Web UI" section) — only trip CHAT (a
/// `ChatScreen(roomId: "trip:<tripId>")` room, reached from the Chat tab's
/// room list) and trip CALENDAR events (merged into the existing native
/// calendar via `GET /api/calendar/events`) are native. This tab is a thin
/// `HybridWebView` host.
struct TripsScreen: View {
    @Environment(AppStore.self) private var store

    var body: some View {
        HybridWebView(path: "/trips", isEmbedded: true)
            // `HybridWebView` sets `contentInsetAdjustmentBehavior = .always`,
            // so — the same trick `SurfaceScaffold` uses via `.contentMargins`
            // (SurfaceScaffold.swift:41) — a bottom `safeAreaInset` here becomes
            // a matching `contentInset` on the WKWebView's own scroll view,
            // keeping page content clear of the floating tab bar without
            // shrinking the visible viewport.
            .safeAreaInset(edge: .bottom, spacing: 0) {
                Color.clear.frame(height: Layout.tabBarClearance)
            }
            // Trip calendar rows are synthesized by the server rather than
            // persisted as editable appointments. The embedded web surface can
            // create, edit, or delete Trips without touching native AppStore
            // state, so refresh that projection whenever the user leaves Trips.
            .onDisappear {
                Task { await store.loadCalendarAndHomework() }
            }
    }
}

File: android/app/src/main/java/com/fametc/app/ui/features/planning/TripsScreen.kt
package com.fametc.app.ui.features.planning

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Scaffold
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.fametc.app.ui.components.HybridWebView
import com.fametc.app.ui.theme.FamTheme

/**
 * Trips webview surface (matches iOS TripsScreen.swift).
 * Itinerary, flights, lodging, invites, and trips list live on the web at /trips.
 */
@Composable
fun TripsScreen() {
    Scaffold(containerColor = FamTheme.colors.bg) { padding ->
        Box(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .padding(bottom = 64.dp)
        ) {
            HybridWebView(path = "/trips", isEmbedded = true)
        }
    }
}
